.github/actions/use-pre-commit/action.yml

```diff
@@ -30,29 +30,23 @@ runs:
         key: pre-commit-${{ inputs.version }}-${{ hashFiles(inputs.config-file) }}
         path: |
           ~/.cache/pre-commit
-          ~/.pre-commit-venv
-
-    - name: Create virtualenv
-      if: steps.cache-pre-commit.outputs.cache-hit != 'true'
-      run: python -m venv ~/.pre-commit-venv
-      shell: bash
 
     - name: Install pre-commit
       if: steps.cache-pre-commit.outputs.cache-hit != 'true'
       run: |
-        source ~/.pre-commit-venv/bin/activate
-        python -m pip install pre-commit==2.19.0
+        set -eux
+        curl --proto '=https' --tlsv1.2 -sSL https://github.com/pre-commit/pre-commit/releases/download/v${{ inputs.version }}/pre-commit-${{ inputs.version }}.pyz > ~/.cache/pre-commit/pre-commit.pyz
       shell: bash
 
     - name: Debug installed python package
       run: |
-        source ~/.pre-commit-venv/bin/activate
-        python -m pip freeze --local
-        pre-commit --version
+        set -eux
+        python --version
+        python ~/.cache/pre-commit/pre-commit.pyz --version
       shell: bash
 
     - name: Run pre-commit
       run: |
-        source ~/.pre-commit-venv/bin/activate
-        pre-commit run --show-diff-on-failure --color=always ${{ inputs.extra-args }}
+        set -eux
+        python ~/.cache/pre-commit/pre-commit.pyz run --show-diff-on-failure --color=always ${{ inputs.extra-args }}
```

Comment:
```diff
-      run: |
-        set -eux
-        python ~/.cache/pre-commit/pre-commit.pyz run --show-diff-on-failure --color=always ${{ inputs.extra-args }}
+      run: python ~/.cache/pre-commit/pre-commit.pyz run --show-diff-on-failure --color=always ${{ inputs.extra-args }}
```

`Github Action` show the first line of `run` since you have 1 command here (excluding the `set ...`) you can put directly `python ~/.cache[...]` and skipping `set ...`